gitlet.js

```diff
@@ -1746,7 +1747,12 @@ var files = {
   // the tree.  `tree` format is: `{ a: { b: { c: "filecontent" }}}`
   writeFilesFromTree: function(tree, prefix) {
     Object.keys(tree).forEach(function(name) {
-      var path = nodePath.join(prefix, name);
+      var path;
+      if (prefix == "/" && process.platform == "win32") {
+        path = nodePath.join(name, prefix);
+      } else {
+        path = nodePath.join(prefix, name);
+      }
```

Comment:
My hunch is that the deep problem here is that Windows file roots look file `E:\` whereas Unix file roots look like `/`.  I'd like to fix this issue with something like the code I pushed to the `fix-windows-root-directory` branch - a function whose job it is to get the root path of the system.  I see that my first attempt didn't solve this problem.  I'm going to have another go.